.github/workflows/build.yml

```diff
@@ -10,12 +10,19 @@ jobs:
   spotless:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
       - name: Spotless Check
         run: ./gradlew spotlessCheck
+  javadoc:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Javadoc CheckStyle
+        run: ./gradlew checkstyleMain
```

Comment:
Should we create a separate task for Checkstyle and not have it under javaDoc?

Reply:
The only thing we are using checkstyle for right now is javadocs. So for now I think it's fine here.  

If we decide to use checkstyle for any other code hygiene stuff I'm fine with splitting it out, but I'd prefer to wait until we actually do that (e.g., star imports).